src/screens/CardScreen.tsx

```diff
@@ -0,0 +1,339 @@
+                    </View>
+                    <View style={styles.showCard}>
+                      <CustomButton onPress={() => alert('Work in progress')} title='Show Card' />
+                    </View>
+                  </>
+                )}
+              </>
+            )}
+            {cardInfoState === 'benefits' && (
+              <>
+                <View style={styles.benefitsContainer}>
+                  <Text style={styles.text}>List of benefits</Text>
+                  {Card.isAdded && (
+                    <BenefitList
+                      benefits={Card.content.benefits}
+                      setBenefit={setBenefit}
+                      setState={setScreen}
+                      customBenefitTileStyle={{ width: '100%', height: 60 }}
+                      mode='addToInventory'
+                    />
+                  )}
+                  {!Card.isAdded && (
+                    <BenefitList
+                      benefits={Card.content.benefits}
+                      onPress={() => {}}
+                      customBenefitTileStyle={{ width: '100%', height: 60 }}
+                      mode='addToInventory'
+                    />
+                  )}
+                </View>
+                {Card.isAdded && (
+                  <View style={[styles.buttonsContainer, { marginTop: '10%' }]}>
+                    <CustomButton
+                      onPress={() => OnCancel()}
+                      title='Cancel'
+                      customButtonStyle={styles.button}
+                    />
+                    <CustomButton
+                      onPress={() => OnSave()}
+                      title='Save'
+                      customButtonStyle={styles.button}
+                    />
+                  </View>
+                )}
+              </>
+            )}
+          </View>
+          {!Card.isAdded && (
+            <CustomButton title='add card' onPress={() => alert('Work in progress')} />
+          )}
+          <TapBar navigation={navigation} />
+        </>
+      )}
+      {screenState === 'benefit' && (
+        <>
+          <TopBar
+            iconLeft='arrow-left'
+            onPressLeft={() => {
+              setInfo('benefits');
+              setScreen('card');
+            }}
+          />
+          <BoxContainer style={styles.benefitDesc}>
+            <Text style={styles.description}>{benefit.description}</Text>
+          </BoxContainer>
+          <CustomButton
+            title='add benefit'
+            onPress={() => AddBenefit(benefit)}
+            customButtonStyle={styles.benefitButton}
+          />
+        </>
+      )}
+      {screenState === 'claimBenefits' && (
+        <>
+          <TopBar
+            iconLeft='arrow-left'
+            onPressLeft={() => {
+              setInfo('benefits');
+              setScreen('card');
+            }}
+          />
+          {/* todo
+          <BenefitList
+            benefits={Card.content.inventory}
```

Comment:
chyba adapter bedzie potrzebny bo mismatch typow 
![1067860950700531842](https://github.com/StampWallet/stamp-wallet/assets/93001484/f7eefde0-e535-4951-a37b-5c6abd75c4c4)

Reply:
albo calkowite przerobienie komponentu xd

Reply:
Jak duży jest mismatch? są po prostu dodatkowe pola?
Jak tak, to wystarczy zrobić typ, który `extends Benefit` czy coś w tym stylu

Reply:
Albo wykorzystać unię